docs/syntax/Cadence.syntax

```diff
@@ -0,0 +1,1482 @@
+    ;
+
+destroyFunctionDeclaration:
+        access *1viewKeyword destroyKeyword parameterList *1functionBlock
+    ;
+
+prepareFunctionDeclaration:
+        access *1viewKeyword prepareKeyword parameterList *1functionBlock
+    ;
+
+functionDeclaration:
+        access *1viewKeyword funKeyword nonReservedIdentifier parameterList *1functionDeclarationTypeAnnotation *1functionBlock
+    ;
+
+functionDeclarationTypeAnnotation:
+        ":" typeAnnotation
+    ;
+
+eventDeclaration:
+        access eventKeyword nonReservedIdentifier parameterList
+    ;
+
+comma:
+        ","
+    ;
+
+pragmaDeclaration:
+        "#" expression
+    ;
+
+entitlementDeclaration:
+        access entitlementKeyword nonReservedIdentifier
+    ;
+
+entitlementMappingDeclaration:
+        access entitlementKeyword mappingKeyword nonReservedIdentifier "{" entitlementMappingsAndInclusions "}"
+    ;
+
+entitlementMappingsAndInclusions:
+        *entitlementMappingElement
+    ;
+
+entitlementMappingElement:
+        entitlementMapping
+    |   entitlementIncludeType
+    ;
+
+entitlementMapping:
+        entitlementMappingSource "->" entitlementMappingTarget
+    ;
+
+entitlementMappingSource:
+        fullType
+    ;
+
+entitlementMappingTarget:
+        fullType
+    ;
+
+entitlementIncludeType:
+        includeKeyword nominalType
+    ;
+
+semicolonEnumerationDeclarationCase:
+        semicolon
+    |   enumerationDeclarationCase
+    ;
+
+enumerationDeclarationCase:
+        access caseKeyword nonReservedIdentifier
+    ;
+
+parameterList:
+        "(" *parameter( "," ) *1comma ")"
+    ;
+
+parameter:
+        parseLabeledParameter
+    |   parseUnlabeledParameter
+    ;
+
+parseLabeledParameter:
+        parameterPublicName parameterLocalName ":" typeAnnotation *1defaultValue
+    ;
+
+parseUnlabeledParameter:
+        parameterLocalName ":" typeAnnotation *1defaultValue
+    ;
+
+parameterPublicName:
+        nonReservedIdentifier
+    ;
+
+parameterLocalName:
+        nonReservedIdentifier
+    ;
+
+defaultValue:
+        "=" expression
+    ;
+
+typeAnnotation:
+        *1resourceAnnotation fullType
+    ;
+
+resourceAnnotation:
+        "@"
+    ;
+
+fullType:
+        fullTypeNormal
+    |   fullTypeNested
+    ;
+
+fullTypeNormal:
+        *1fullTypeReference innerType *fullTypeOptional
+    ;
+
+fullTypeReference:
+        *1fullTypeReferenceAuth "&"
+    ;
+
+fullTypeReferenceAuth:
+        fullTypeReferenceAuthConjunction
+    |   fullTypeReferenceAuthDisjunction
+    |   fullTypeReferenceAuthNoEntitlements
+    |   fullTypeReferenceAuthMapping
+    ;
+
+fullTypeReferenceAuthConjunction:
+        authKeyword "(" 1*fullType( "," ) ")"
+    ;
+
+fullTypeReferenceAuthDisjunction:
+        authKeyword "(" 1*fullType( "|" ) ")"
+    ;
+
+fullTypeReferenceAuthNoEntitlements:
+        authKeyword
+    ;
+
+fullTypeReferenceAuthMapping:
+        authKeyword "(" mappingKeyword nominalType ")"
+    ;
+
+fullTypeOptional:
+        . optional
```

Comment:
The `.` looks like "any character" here, like in other notations. Maybe leave a comment

Reply:
The `.` and `_` were chosen to represent adjacent and not-adjacent. The language for the parser section does not match individual characters. It matches tokens.

Reply:
Right, that's OK. It might not be obvious to readers unfamiliar with this custom notation, and given the `.` notation means something else in other commonly used notations, maybe help the reader here and point out that this does not mean "any character" but "adjacent"